demos/http/http_demo_s3_download/http_demo_s3_download.c

```diff
@@ -632,49 +647,71 @@ int main( int argc,
-    }
+        /* Define the transport interface. */
+        if( returnStatus == EXIT_SUCCESS )
+        {
+            ( void ) memset( &transportInterface, 0, sizeof( transportInterface ) );
+            transportInterface.recv = Openssl_Recv;
+            transportInterface.send = Openssl_Send;
+            transportInterface.pNetworkContext = &networkContext;
+        }
 
-    /******************** Download S3 Object File. **********************/
+        /******************** Download S3 Object File. **********************/
 
-    if( returnStatus == EXIT_SUCCESS )
-    {
-        /* Retrieve the path location from S3_PRESIGNED_GET_URL. This
-         * function returns the length of the path without the query into
-         * pathLen, which is left unused in this demo. */
-        httpStatus = getUrlPath( S3_PRESIGNED_GET_URL,
-                                 S3_PRESIGNED_GET_URL_LENGTH,
-                                 &pPath,
-                                 &pathLen );
-
-        returnStatus = ( httpStatus == HTTPSuccess ) ? EXIT_SUCCESS : EXIT_FAILURE;
-    }
+        if( returnStatus == EXIT_SUCCESS )
+        {
+            /* Retrieve the path location from S3_PRESIGNED_GET_URL. This
+             * function returns the length of the path without the query into
+             * pathLen, which is left unused in this demo. */
+            httpStatus = getUrlPath( S3_PRESIGNED_GET_URL,
+                                     S3_PRESIGNED_GET_URL_LENGTH,
+                                     &pPath,
+                                     &pathLen );
+
+            returnStatus = ( httpStatus == HTTPSuccess ) ? EXIT_SUCCESS : EXIT_FAILURE;
+        }
 
-    if( returnStatus == EXIT_SUCCESS )
-    {
-        ret = downloadS3ObjectFile( &transportInterface,
-                                    pPath );
-        returnStatus = ( ret == true ) ? EXIT_SUCCESS : EXIT_FAILURE;
-    }
+        if( returnStatus == EXIT_SUCCESS )
+        {
+            ret = downloadS3ObjectFile( &transportInterface,
+                                        pPath );
+            returnStatus = ( ret == true ) ? EXIT_SUCCESS : EXIT_FAILURE;
+        }
+
+        /************************** Disconnect. *****************************/
+
+        /* End the TLS session, then close the TCP connection. */
+        ( void ) Openssl_Disconnect( &networkContext );
+
+        /******************* Retry in case of failure. **********************/
+
+        /* Increment the demo run count. */
+        demoRunCount++;
+
+        if( returnStatus == EXIT_SUCCESS )
+        {
+            LogInfo( ( "Demo iteration %d is successful.", demoRunCount ) );
+        }
+        /* Attempt to retry a failed iteration of demo for up to #HTTP_MAX_DEMO_COUNT times. */
+        else if( demoRunCount < HTTP_MAX_DEMO_COUNT )
+        {
+            LogWarn( ( "Demo iteration %d failed. Retrying...", demoRunCount ) );
+            sleep( DELAY_BETWEEN_DEMO_ITERATIONS_S );
```

Comment:
Might it be better to use `Clock_SleepMs` instead?

Reply:
Why are you suggesting that? 
I think it is better to use sleep directly here as the time unit we are using is seconds.

Reply:
Okay just a nit from me. I'm guessing if we have platform-specific sleep then it will be easier to port demos to FreeRTOS.